cloudant-spring-boot-starter/build.gradle

```diff
@@ -12,7 +12,23 @@
  * and limitations under the License.
  */
 
+
+apply plugin: 'propdeps'
+apply plugin: 'propdeps-maven'
+
+buildscript {
+    repositories {
+        maven { url 'http://repo.spring.io/plugins-release' }
+    }
+    dependencies {
+        classpath 'io.spring.gradle:propdeps-plugin:0.0.9.RELEASE'
+    }
+}
+
 dependencies {
   compile group: 'org.springframework.boot', name: 'spring-boot', version: springBootVersion
   compile group: 'org.springframework.boot', name: 'spring-boot-autoconfigure', version: springBootVersion
+  optional group: 'org.springframework.boot', name: 'spring-boot-configuration-processor', version: springBootVersion
 }
+
+compileJava.dependsOn(processResources)
```

Comment:
Adding this as a dependency to `compileJava` seems an odd choice to me. From https://github.com/spring-gradle-plugins/propdeps-plugin it sounds like this runs as part of `install`. We don't run `install` directly, but it might be part of the task graph for `publish`. Did you have issues getting it to generate as part of `install`?

Reply:
The reason I added it as a dependency for compileJava was due to the comment here: https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#configuration-metadata-annotation-processor
"You need to add compileJava.dependsOn(processResources) to your build to ensure that resources are processed before code is compiled. Without this directive any additional-spring-configuration-metadata.json files will not be processed."
So if wanted to add non-generated spring metadata files in the future we can without changing the build script.

Reply:
Since we aren't providing an additional one yet we could leave this out for now

Reply:
No, it's fine that link explains why it needs to be before the compile stage.